libibverbs/device.c

```diff
@@ -262,6 +267,8 @@ struct ibv_context *__ibv_open_device(struct ibv_device *device)
 	context->cmd_fd = cmd_fd;
 	pthread_mutex_init(&context->mutex, NULL);
 
+	ibverbs_device_hold(device);
+
```

Comment:
I feel like this should fail early here if the device is already known to be deleted and not grab another ref on it. eg consider the case where the dev node has been reclaimed by another device. There is a race here of some kind with the kernel. We need to do design this so that the kernel cannot re-use the cdev numbers or sysfs file names, and do the timestamp test appropriately to ensure that we are not opening the wrong instance of a ibv_device.